core/src/main/scala/org/apache/spark/partial/SumEvaluator.scala

```diff
@@ -56,9 +56,12 @@ private[spark] class SumEvaluator(totalOutputs: Int, confidence: Double)
       val confFactor = {
         if (counter.count > 100) {
           new NormalDistribution().inverseCumulativeProbability(1 - (1 - confidence) / 2)
-        } else {
+        } else if (counter.count > 1) {
           val degreesOfFreedom = (counter.count - 1).toInt
           new TDistribution(degreesOfFreedom).inverseCumulativeProbability(1 - (1 - confidence) / 2)
+        } else {
+          // No way to meaningfully estimate confidence, so we signal no particular confidence interval
+          Double.PositiveInfinity
```

Comment:
I'm not sure I follow you. If `sumStdev` is `NaN` it really won't matter what this value is. If it's any other value, multiplying by `Infinity` results in `Infinity`, which I believe is what is desired. 

`sumStdDev` will only be `NaN` if `sumVar` is negative. By the construction of `sumVar`, I don't think that's possible. 

```
scala> Double.PositiveInfinity * 2
res1: Double = Infinity

scala> Double.PositiveInfinity * 2.0
res2: Double = Infinity

scala> Double.PositiveInfinity * Double.PositiveInfinity
res3: Double = Infinity

scala> Double.PositiveInfinity * Double.NegativeInfinity
res4: Double = -Infinity

```